src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs

```diff
@@ -246,6 +259,8 @@ public ActionEndpointFactory(RoutePatternTransformer routePatternTransformer, IE
             route.Conventions[i](builder);
         }
 
+        builder.FilterFactories.Clear();
```

Comment:
Depending on the merge order, either this PR or my PR (#42827) will want to undue this because my PR stops the re-application of filter during Build().

I'm okay merging second, so this is really a note to self.

Reply:
I'm setting #42827 to auto-merge, so it's likely this PR that will have the merge conflict. But it means we can remove this line:

```suggestion
```